File: src/main/scala/evan/wang/flink/examples/utils/PropertiesUtil.scala
package evan.wang.flink.examples.utils

import java.util.Properties

import org.apache.commons.collections.CollectionUtils
import org.slf4j.LoggerFactory

object PropertiesUtil {
  private val logger = LoggerFactory.getLogger("Property")
  private val contextProperties = new Properties()
  try {
    val inputStreamCommon = Thread.currentThread().getContextClassLoader.getResourceAsStream("common.properties") //文件要放到resource文件夹下
    contextProperties.load(inputStreamCommon)
    if (CollectionUtils.isEmpty(contextProperties.entrySet())) throw new IllegalArgumentException("读取不到配置信息")
    logger.info("读取配置文件结束")
  } catch {
    case exception: Exception => logger.error("读取配置文件异常", exception)
  }

  def getIntValue(key: String): Int = {
    val strValue = getStrValue(key)
    strValue.toInt
  }

  def getIntValue(key: String, defaultValue: Int): Int = {
    val strValue = getStrValue(key)
    if (strValue == null) {
      defaultValue
    } else
      strValue.toInt
  }

  def getStrValue(key: String): String = contextProperties.getProperty(key)

  def getKafkaProperties(groupId: String): Properties = {
    val properties = new Properties
    properties.setProperty("bootstrap.servers", getStrValue("kafka.bootstrap.servers"))
    properties.setProperty("zookeeper.connect", getStrValue("kafka.zookeeper.connect"))
    properties.setProperty("group.id", groupId)
    properties
  }

}

File: src/main/java/evan/wang/flink/examples/utils/PropertiesUtil.java
package evan.wang.flink.examples.utils;

import org.apache.commons.collections.CollectionUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.InputStream;
import java.util.Properties;

public class PropertiesUtil {
    private static final Logger logger = LoggerFactory.getLogger(PropertiesUtil.class);
    private static final Properties contextProperties = new Properties();
    static {
        //文件要放到resource文件夹下
        try(InputStream inputStreamCommon = Thread.currentThread().getContextClassLoader().getResourceAsStream("common.properties")){
            contextProperties.load(inputStreamCommon);
            if (CollectionUtils.isEmpty(contextProperties.entrySet())) throw new IllegalArgumentException("读取不到配置信息");
            logger.info("读取配置文件结束");
        }catch (Exception e){
            logger.error(e.getMessage(), e);
        }
    }

    public static Integer  getIntValue(String key){
        return Integer.parseInt(getStrValue(key));
    }

    public static Integer  getIntValue(String key, int defaultValue)  {
        String strValue = getStrValue(key);
        if (strValue == null) {
           return defaultValue;
        } else{
            return Integer.parseInt(strValue);
        }
    }

    public static String  getStrValue(String key){
       return contextProperties.getProperty(key);
    }

    public static Properties getKafkaProperties(String groupId){
        Properties properties = new Properties();
        properties.setProperty("bootstrap.servers", getStrValue("kafka.bootstrap.servers"));
        properties.setProperty("zookeeper.connect", getStrValue("kafka.zookeeper.connect"));
        properties.setProperty("group.id", groupId);
        return properties;
    }


}
